test/core/stream/batch_t.cpp

```diff
@@ -29,6 +29,18 @@ BOOST_AUTO_TEST_CASE(test_batch_insert_assign_extract)
     BOOST_CHECK_THROW(batch.extract<Double>(), std::runtime_error);
 }
 
+BOOST_AUTO_TEST_CASE(test_batch_insert_at_raw_insert)
+{
+    using hipipe::stream::batch_t;
+
+    batch_t batch, batch2;
+    batch.insert_or_assign<Int>(std::vector<int>{0, 1, 2});
+    BOOST_TEST(batch.at<Int>()->extract<Int>() == std::vector<int>({0, 1, 2}));
+    BOOST_CHECK_THROW(batch.at<Double>(), std::runtime_error);
+
+    batch2.raw_insert_or_assign<Int>(std::move(batch.at<Int>()));
+    BOOST_TEST(batch2.at<Int>()->extract<Int>() == std::vector<int>({0, 1, 2}));
+}
```

Comment:
I have to admit I would prefer to test those functions separately as everywhere in this project where it is possible. But not that much to let you rewrite it.